### DISCUSSION TITLE
How to lazy load a model/EmbeddingFunction

### DISCUSSION THREAD

**Discussion by humpydonkey** [lancedb/lancedb#1171] [Q&A]
Hi team,

Just a QQ: is there way to lazy load a model that is used as `EmbeddingFunction`?

More context: loading a sizable model into memory could take a couple seconds or more.
Just wondering if there is a way to lazy load a model when defining a LanceDB schema, e.g. the `Image` class below.

```
import torch
from lancedb.embeddings import EmbeddingFunctionRegistry
from lancedb.pydantic import LanceModel, Vector

registry = EmbeddingFunctionRegistry.get_instance()
clip = registry.get("open-clip").create(
    name="ViT-H-14-378-quickgelu",
    pretrained="dfn5b",
    device="cuda" if torch.cuda.is_available() else "cpu",
)

class Image(LanceModel):
    timestamp: float
    uri: str = clip.SourceField()
    vec_clip: Vector(clip.ndims()) = clip.VectorField()
```

**Comment by AyushExel** [comment]
Its not really lazyloading, but internally most of the available embedding functions cache the model so it's only loaded once in memory per instance.
